test/helper.cpp

```diff
@@ -39,13 +39,27 @@ int main()
     TEST(fabs(wrap_2pi(-201.) - (-201. + 32. * M_TWOPI)) < FLT_EPSILON);
     TEST(!is_finite(wrap_2pi(NAN)));
 
+    // Equality checks
+    TEST(isEqualF(1., 1.));
+    TEST(!isEqualF(1., 2.));
+    TEST(!isEqualF(NAN, 1.f));
+    TEST(!isEqualF(1.f, NAN));
+    TEST(!isEqualF(INFINITY, 1.f));
+    TEST(!isEqualF(1.f, INFINITY));
+    TEST(!isEqualF(NAN, NAN));
+    TEST(!isEqualF(INFINITY, INFINITY));
+
     Vector3f a(1, 2, 3);
     Vector3f b(4, 5, 6);
     TEST(!isEqual(a, b));
     TEST(isEqual(a, a));
 
-    TEST(isEqualF(1., 1.));
-    TEST(!isEqualF(1., 2.));
+    Vector3f c(1, 2, 3);
+    Vector3f d(1, 2, NAN);
+    TEST(!isEqual(c, d));
+    TEST(isEqual(c, c));
+    TEST(!isEqual(d, d));
```

Comment:
Good point. I considered it and then thought we are "forced by convention" to follow floating-point arithmetic. But it would simplify client code a lot if you could use the equal check for NAN as well if you expect it as output.

Reply:
https://github.com/PX4/Matrix/pull/93/commits/2fe6ec655d019382c327529f06bc7d5abfb0f787